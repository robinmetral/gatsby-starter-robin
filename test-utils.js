// wraps all tests with the emotion theme provider
// from https://testing-library.com/docs/react-testing-library/setup#custom-render

import React from "react"
import { render } from "@testing-library/react"
import { ThemeProvider } from "emotion-theming"

import theme from "./src/theme"

const ThemeWrapper = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

const customRender = (ui, options) =>
  render(ui, { wrapper: ThemeWrapper, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
