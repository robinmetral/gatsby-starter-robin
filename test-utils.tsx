// wraps all tests with the emotion theme provider
// from https://testing-library.com/docs/react-testing-library/setup#custom-render

import React, { ReactNode, ReactElement } from "react"
import { render } from "@testing-library/react"
import { ThemeProvider } from "emotion-theming"
import { CacheProvider } from "@emotion/core"
import createCache from "@emotion/cache"

import theme from "./src/theme"
import GlobalStyles from "./src/components/GlobalStyles"

/* create emotion cache to avoid :first-child warnings */
const emotionCache = createCache()
emotionCache.compat = true

type TestWrapperProps = {
  children?: ReactNode
}

const TestWrapper = ({ children }: TestWrapperProps) => {
  return (
    <CacheProvider value={emotionCache}>
      <ThemeProvider theme={theme}>
        <GlobalStyles />
        {children}
      </ThemeProvider>
    </CacheProvider>
  )
}

const customRender = (ui: ReactElement, options?: object) =>
  render(ui, { wrapper: TestWrapper, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
