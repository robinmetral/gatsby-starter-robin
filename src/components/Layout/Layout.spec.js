import React from "react"
import { useStaticQuery } from "gatsby"
import { render } from "test-utils"

import Layout from "./Layout"
import layoutStaticQuery from "./__fixtures__/layoutStaticQuery"

beforeEach(() => {
  useStaticQuery.mockImplementation(() => layoutStaticQuery)
})

const mockPageContext = {
  frontmatter: {
    title: "Home",
  },
}

describe("Layout", () => {
  it("should render its content", () => {
    const { getByTestId } = render(
      <Layout pageContext={mockPageContext}>
        <div data-testid="content">Page content</div>
      </Layout>
    )

    expect(getByTestId("content")).toHaveTextContent("Page content")
  })

  it("should match the snapshots", () => {
    const layout = render(
      <Layout pageContext={mockPageContext}>
        <div>Page content</div>
      </Layout>
    )

    expect(layout).toMatchSnapshot()
  })
})
