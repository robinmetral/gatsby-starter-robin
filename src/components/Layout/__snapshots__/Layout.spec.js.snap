// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should match the snapshots 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .emotion-4 {
  background: rebeccapurple;
}

.emotion-2 {
  margin: 0 auto;
  max-width: 960px;
  padding: 1.45rem;
}

.emotion-1 {
  margin: 0;
}

.emotion-0 {
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<body>
    <div>
      <header
        class="emotion-4"
      >
        <div
          class="emotion-2 emotion-3"
        >
          <h1
            class="emotion-1"
            data-testid="title"
          >
            <a
              class="emotion-0"
              href="/"
            >
              Gatsby Default Starter
            </a>
          </h1>
        </div>
      </header>
      <div
        class="emotion-2 emotion-3"
      >
        <main>
          <div>
            Page content
          </div>
        </main>
        <footer>
          © 
          2019
          , Built with
           
          <a
            href="https://www.gatsbyjs.org"
          >
            Gatsby
          </a>
        </footer>
      </div>
    </div>
  </body>,
  "container": .emotion-4 {
  background: rebeccapurple;
}

.emotion-2 {
  margin: 0 auto;
  max-width: 960px;
  padding: 1.45rem;
}

.emotion-1 {
  margin: 0;
}

.emotion-0 {
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<div>
    <header
      class="emotion-4"
    >
      <div
        class="emotion-2 emotion-3"
      >
        <h1
          class="emotion-1"
          data-testid="title"
        >
          <a
            class="emotion-0"
            href="/"
          >
            Gatsby Default Starter
          </a>
        </h1>
      </div>
    </header>
    <div
      class="emotion-2 emotion-3"
    >
      <main>
        <div>
          Page content
        </div>
      </main>
      <footer>
        © 
        2019
        , Built with
         
        <a
          href="https://www.gatsbyjs.org"
        >
          Gatsby
        </a>
      </footer>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
